LCLSGeneral/LCLSGeneral/POUs/Hardware/FB_LED.TcPOU

```diff
@@ -0,0 +1,62 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<TcPlcObject Version="1.1.0.1" ProductVersion="3.1.4022.18">
+  <POU Name="FB_LED" Id="{5fae4b24-b1d6-4ac5-a010-9aff4e2893db}" SpecialFunc="None">
+    <Declaration><![CDATA[FUNCTION_BLOCK FB_LED
+(*
+    Reads a percentage of illumination determined by the user and converts it to a raw value
+    to send to the terminal output via the FB_AnalogOutput.
+	
+	The determination of the full scale raw value (FSV) is calculated via the iTermBits,
+    which is setup as an input variable to be set according to the situation. This value
+    is bounded by the FSV of the terminal set by the Beckhoff architecture: (2^15 - 1) = 32767.  
+	
+	Through the configurable max raw value sent to the terminal output, this LED function block
+    is applicable to a wider variety of hardware: whether it be simple Analog Voltage control,
+    PWM voltage output, or current controlled output. 
+
+    As an initial application, this FB will be instituted for systems of vaccum chamber LED 
+    illuminators on TMO endstations, which are rated for 12 V max. They will be dimmed via the
+    EL2502 terminal 24 V PWM output voltage control from 0-50% duty cycle.
+	
+    2022-5-03 Maarten Thomas-Bosum
+*)
+VAR_INPUT
+	iTermBits : UINT;
+END_VAR
+VAR
+	iIlluminatorINT AT %Q*: INT;
+
+    {attribute 'pytmc' := '
+        pv: PWR
+        field: ZNAM OFF
+        field: ONAM ON
+    '}
+    bLedPower AT %Q*: BOOL;
+
+    {attribute 'pytmc' := '
+        pv: CIL:PCT
+        field: EGU %
+    '}
+    fIlluminatorPercent: LREAL;
```

Comment:
One last small thing to consider: PV names
PWR = power, makes sense
CIL:PCT = camera illuminator, colon, percent
Is the CIL part intended or left over from a copy/paste?

Reply:
I think it is intended. Other LED illuminators have the CIL fungible term included in their PV names.

Reply:
Okay, if you intend for this FB to only be used for cameras, then that's ok. Just make sure to add that to the docs.

Reply:
I changed this to "ILL:PCT" now